### ISSUE TITLE
Crash when syncing item with "%0A"

### ISSUE THREAD

**Issue by WhyNotHugo** [pimutils/vdirsyncer#918]
With the current master I'm seeing an issue with a specific item. Given the URL, I'm suspecting an encoding issue or something funny with that:

```
error: Unknown error occurred for calendars/502708f4-4b94-4963-b042-f7340c829bdf: /dav/calendars/user/whynothugo@fastmail.com/502708f4-4b94-4963-b042-f7340c829bdf/UID:20210609T084907Z-@synaps-web-54fddfdf7-7kcfm%0A.ics
error: Use `-vdebug` to see the full traceback.
debug:   File "/usr/lib/python3.9/site-packages/vdirsyncer/cli/tasks.py", line 72, in sync_collection
debug:     await sync.sync(
debug:   File "/usr/lib/python3.9/site-packages/vdirsyncer/sync/__init__.py", line 145, in sync
debug:     b_nonempty = await b_info.prepare_new_status()
debug:   File "/usr/lib/python3.9/site-packages/vdirsyncer/sync/__init__.py", line 62, in prepare_new_status
debug:     async for href, item, etag in self.storage.get_multi(prefetch):
debug:   File "/usr/lib/python3.9/site-packages/vdirsyncer/storage/dav.py", line 547, in get_multi
debug:     raise exceptions.NotFoundError(href)
```

The way this raises also makes syncing stop immediately, since it closes the connector. Syncing should continue.

**Comment by WhyNotHugo** [comment]
Well, the UID contains a newline character:

```
In [2]: urllib.parse.unquote("%0A")
Out[2]: '\n'
```

This events are generated by coronatest.nl.

**Comment by WhyNotHugo** [comment]
Skipping normalisation makes it work, but breaks other things.

**Comment by WhyNotHugo** [comment]
Fixed via #929
